legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/tools/SessionProvider.java

```diff
@@ -16,18 +16,59 @@
 
 import org.finos.legend.sdlc.server.auth.LegendSDLCWebFilter;
 import org.finos.legend.sdlc.server.auth.Session;
+import org.finos.legend.sdlc.server.gitlab.GitLabAppInfo;
+import org.finos.legend.sdlc.server.gitlab.auth.GitLabSessionBuilder;
+import org.finos.legend.server.pac4j.gitlab.GitlabClient;
+import org.pac4j.core.context.J2EContext;
+import org.pac4j.core.context.Pac4jConstants;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.context.session.SessionStore;
+import org.pac4j.core.profile.CommonProfile;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletRequestWrapper;
 import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
+import java.util.LinkedHashMap;
 
-public class SessionUtil
+public class SessionProvider
 {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(SessionUtil.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(SessionProvider.class);
+
+    private SessionStore sessionStore;
+
+    public Session getSessionFromSessionStore(HttpServletRequest httpRequest, HttpServletResponse httpResponse, GitLabAppInfo appInfo)
+    {
+        if (sessionStore != null)
+        {
+            WebContext context = new J2EContext(httpRequest, httpResponse);
+
+            LinkedHashMap<String, CommonProfile> map = (LinkedHashMap) sessionStore.get(context, Pac4jConstants.USER_PROFILES);
+
+            if (map != null && map.containsKey(GitlabClient.GITLAB_CLIENT_NAME))
+            {
+                CommonProfile profile = map.get(GitlabClient.GITLAB_CLIENT_NAME);
+                GitLabSessionBuilder builder = GitLabSessionBuilder.newBuilder(appInfo);
+                return builder.withProfile(profile).build();
+            }
```

Comment:
Even if the map contains the key, the value could still be null. So we need to check for that.
```suggestion
            if (map != null))
            {
                CommonProfile profile = map.get(GitlabClient.GITLAB_CLIENT_NAME);
                if (profile != null)
                {
                    return GitLabSessionBuilder.newBuilder(appInfo).withProfile(profile).build();
                }
            }
```

Reply:
done